### PR TITLE
Added 'Manifest V2 extensions' section in settings.

### PR DESCRIPTION
**Pull request by boocmp**
Resolves https://github.com/brave/brave-browser/issues/28367

https://user-images.githubusercontent.com/73575789/216320363-1b325600-b7db-4aa7-a68c-dde06572445c.mp4

image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Launch Brave with ```ExtensionsManifestV2``` feature enabled.
2. Go to brave://settings/extensions
3. Click to ```Manifest V2 Extensions``` section
4. You should see 4 extensions
5. Enable/Disable toggles will isntall/uninstall corresponding extensions

### CODE REVIEW

**Comment by iefremov** [comment]
pls file a ticket with a link to the spec

**Review by antonok-edm — COMMENTED** [review]
We probably want another review for the strings, but I added my own suggestions.

File: app/brave_settings_strings.grdp [hunk 1]
```diff
@@ -901,6 +901,39 @@
...
+    Manifest V2 extensions
+  </message>
+  <message name="IDS_SETTINGS_MANAGE_EXTENSIONS_V2_SUBLABEL" desc="The sub label of manage v2 extensions link in settings">
+    The following Manifest V2 extensions are no longer available in the Chrome Web Store, but still supported in Brave. Click to download and install.
+  </message>
+  <message name="IDS_SETTINGS_MANAGE_EXTENSIONS_V2_TOAST_CONFIRM" desc="The label of the error toast button in settings">
+    Close
+  </message>
+  <message name="IDS_SETTINGS_MANAGE_EXTENSIONS_V2_NO_SCRIPT_NAME" desc="Manifest V2 extension toggle's label">
+    Enable NoScript&#x2122;
```
[the remark this answers is not in the document]
Comment (boocmp) [reply app/brave_settings_strings.grdp:914]: @rebron asked me to add trademarks. Rafael what do you think?

[the remark this answers is not in the document]
Comment (rebron) [reply app/brave_settings_strings.grdp:914]: Yes, confirmed with @bradleyrichter.

File: app/brave_settings_strings.grdp [hunk 2]
```diff
@@ -901,6 +901,39 @@
...
+    Enable uMatrix&#x2122;
+  </message>
+  <message name="IDS_SETTINGS_MANAGE_EXTENSIONS_V2_UMATRIX_DESC" desc="Manifest V2 extension toggle's sub label">
+    uMatrix is a Manifest V2 extension that provides point-and-click matrix-based firewall, with many privacy-enhancing tools.
+  </message>
+  <message name="IDS_SETTINGS_MANAGE_EXTENSIONS_V2_ADGUARD_NAME" desc="Manifest V2 extension toggle's label">
+    Enable AdBlock&#x2122;
+  </message>
+  <message name="IDS_SETTINGS_MANAGE_EXTENSIONS_V2_ADGUARD_DESC" desc="Manifest V2 extension toggle's sub label">
+    AdGuard is a Manifest V2 extension that effectively blocks all types of ads on all web pages, even on Facebook, YouTube and others.
```
[the remark this answers is not in the document]
Comment (bradleyrichter) [reply app/brave_settings_strings.grdp:935]: TM is useful to identify these extension names as "not Brave's thing" and TM has loose rules for usage. Any company can claim a TM but it carries little value without official registration which then earns you an ®.

File: browser/ui/webui/settings/brave_extensions_manifest_v2_handler.cc [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+void BraveExtensionsManifestV2Handler::NotifyExtensionManifestV2Changed(
+    content::BrowserContext* browser_context,
+    const std::string& id,
+    bool enabled) {
+  if (!IsJavascriptAllowed() ||
+      browser_context != web_ui()->GetWebContents()->GetBrowserContext()) {
+    return;
+  }
+  auto fnd = std::find_if(extensions_.begin(), extensions_.end(),
```
Comment (iefremov) [inline browser/ui/webui/settings/brave_extensions_manifest_v2_handler.cc:166]: base::ranges::find

File: app/brave_settings_strings.grdp [hunk 4]
```diff
@@ -901,6 +901,39 @@
   <message name="IDS_SETTINGS_ENABLE_WIDEVINE_DESC" desc="Text fragment for enabling widevine component. 'Widevine' is the name of a plugin and should not be translated.">
     Use Widevine digital rights management (DRM) component to view protected streaming video and other content in the browser.
   </message>
+  <message name="IDS_SETTINGS_MANAGE_EXTENSIONS_V2_LABEL" desc="The label of manage v2 extensions link in settings">
+    Manifest V2 extensions
+  </message>
+  <message name="IDS_SETTINGS_MANAGE_EXTENSIONS_V2_SUBLABEL" desc="The sub label of manage v2 extensions link in settings">
+    The following Manifest V2 extensions will be removed from the Chrome Web Store, but are still supported in Brave. Click to download and install.
```
Comment (petemill) [inline app/brave_settings_strings.grdp:908]: Should this be "toggle to download and install, or to uninstall"?

File: browser/resources/settings/brave_default_extensions_page/brave_extensions_manifest_v2_subpage.html [hunk 5]
```diff
@@ -0,0 +1,56 @@
+<style include="cr-shared-style">
+  #errorMessage {
+    align-items: center;
```
Comment (petemill) [inline browser/resources/settings/brave_default_extensions_page/brave_extensions_manifest_v2_subpage.html:3]: nit: align-items should be colocated with `display: flex`, `flex-direction`, and `gap` (or anything which affects children's layout

File: browser/resources/settings/brave_default_extensions_page/brave_extensions_manifest_v2_subpage.html [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    background: rgb(255, 225, 147);
+    border-radius: 8px;
+    box-shadow: 0 10px 32px var(--shadow-color);
+    display: flex;
+    flex-direction: row;
+    gap: 16px;
+    left: 50%;
```
Comment (petemill) [inline browser/resources/settings/brave_default_extensions_page/brave_extensions_manifest_v2_subpage.html:10]: nit: `left` should be colocated with the other `position` items

**Review by petemill — APPROVED** [review]
The WebUI Polymer code looks good!

File: app/brave_settings_strings.grdp (hunk 4 above)
Comment (rebron) [reply app/brave_settings_strings.grdp]: cc: @rmcfadden3 Any suggestions on the text. Also, @boocmp We can finalize text with a follow-up. I wouldn't consider this blocking.

Comment (rmcfadden3) [reply app/brave_settings_strings.grdp]: Per Slack convo with @rebron , let's go with this:

`The following Manifest V2 extensions will be removed from the Chrome Web Store, but still be supported in Brave. Toggle on an individual extension to download / install it, or toggle off to uninstall / remove.`

**Pull request merged**